cli.py

```diff
@@ -35,87 +91,103 @@
 trace_logger.addHandler(logging.StreamHandler())
 trace_logger.setLevel(logging.INFO)
 
+def setup_tracing(endpoint: str = None, service_name: str = None):
+    """Setup OpenTelemetry tracing."""
+    try:
+        exporter = OTLPSpanExporter(endpoint=endpoint, insecure=True)
+        provider = TracerProvider(resource=Resource.create({ResourceAttributes.SERVICE_NAME: service_name}))
+        provider.add_span_processor(BatchSpanProcessor(exporter))
+        trace.set_tracer_provider(provider)
+        tracer = trace.get_tracer(service_name)
+        logging.info(f"OpenTelemetry tracing enabled, exporting to {endpoint}")
+        return tracer
+    except Exception as e:
+        logging.error(f"Failed to setup tracing: {e}")
+        return None
+
+
+def _default_to_text(self) -> str:
+    return getattr(self, "content", repr(self))
+BaseMessage.to_text = _default_to_text
 
 def parse_args():
     parser = argparse.ArgumentParser("Sagi CLI")
-    parser.add_argument("--env", type=str, choices=["dev", "prod"], default="dev")
+    parser.add_argument("--env", choices=["dev", "prod"], default="dev")
     parser.add_argument(
-        "--config", type=str, default="src/Sagi/workflows/planning.toml"
+        "--config", default="src/Sagi/workflows/planning.toml"
     )
     parser.add_argument(
         "--trace", action="store_true", help="Enable OpenTelemetry tracing"
     )
     parser.add_argument(
-        "--trace_endpoint",
-        type=str,
-        default="http://localhost:4317",
-        help="OpenTelemetry collector endpoint",
+        "--trace_endpoint", default="http://localhost:4317",
+        help="OpenTelemetry collector endpoint"
+    )
+    parser.add_argument(
+        "--trace_service_name", default="sagi_tracer",
+        help="Service name for OpenTelemetry tracing"
+    )
+    parser.add_argument(
+        "-s", "--session-id", type=str,
+        help="Specify the session ID to load or save; if not provided, one will be generated automatically."
     )
     parser.add_argument(
-        "--trace_service_name",
-        type=str,
-        default="sagi_tracer",
-        help="Service name for OpenTelemetry tracing",
+        "--list-sessions", action="store_true",
+        help="List existing session IDs and exit."
     )
     return parser.parse_args()
 
+async def main_cmd(args: argparse.Namespace):
+    # List sessions and exit
+    if args.list_sessions:
+        conn = get_conn()
+        cur = conn.cursor()
+        cur.execute("SELECT session_id FROM session_states ORDER BY updated_at DESC")
+        rows = cur.fetchall()
+        cur.close(); conn.close()
+        sessions = [r[0] for r in rows]
+        print("Available sessions:", sessions or ["<none>"])
+        return
+
+    session_id = args.session_id or str(uuid.uuid4())
+    print(f"use session_id = {session_id!r}")
```

Comment:
`print` is a little "ugly", maybe use `logging.info` instead
```suggestion
        logging.info("Available sessions:", sessions or ["<none>"])
        return

    session_id = args.session_id or str(uuid.uuid4())
    logging.info(f"use session_id = {session_id!r}")
```